src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentActor.scala

```diff
@@ -176,7 +175,7 @@ private class DeploymentActor(
         tasksToStart.fold(Future.successful(Done)) { tasksToStart =>
           logger.debug(s"Start next $tasksToStart tasks")
           val promise = Promise[Unit]()
-          context.actorOf(childSupervisor(TaskStartActor.props(deploymentManagerActor, status, scheduler, launchQueue, instanceTracker, eventBus,
+          context.actorOf(childSupervisor(TaskStartActor.props(deploymentManagerActor, status, scheduler, launchQueue, eventBus,
```

Comment:
I think you can get rid of the `launchQueue` here - it's needed in the _StartingBehavior_ down the line, which _Scheduler_ makes obsolete now.

Reply:
Yep, that's [phase 3](https://jira.mesosphere.com/browse/MARATHON-8292).

Reply:
I tried to move the launch queue but it's quite a mess https://github.com/mesosphere/marathon/commit/3c01924c32f2a1f6c26659f380aa524a3392411d. Since we are removing the queue in phase  it's not really worth the effort.